### PR TITLE
Add Dockerfile for E2E tests

### PR DESCRIPTION
**Pull request by gigawhitlocks**
#### Summary

Adds a Dockerfile for building an image through which to run the E2E tests in k8s.

#### Ticket Link

Part of [MM-38160](https://mattermost.atlassian.net/browse/MM-38160)

### CODE REVIEW

File: test/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from golang:1.14
```
Comment (cpanato) [inline test/Dockerfile:1]: maybe can use the same go version that is being used in the go mod: https://github.com/mattermost/awat/blob/main/go.mod#L3

also I would use the docker keywords in all caps, like  `FROM`  / `WORKDIR`...

Comment (gigawhitlocks) [reply test/Dockerfile:1]: I always hated the all-caps keywords in Dockerfiles, personally; reminds me of BASIC. I'll change it anyway.

**Comment by gigawhitlocks** [comment]
Force-push is just a fix to the commit message itself; I typed the wrong version into the commit message :facepalm:

**Review by gabrieljackson — APPROVED** [review]
👍

**Review by cpanato — COMMENTED** [review]
/lgtm
thanks!

**Pull request merged**